tests/conftest.py

```diff
@@ -94,51 +89,272 @@ def sat_data():
+            "nwp-ecmwf__latitude": nwp_lat,
+            "nwp-ecmwf__longitude": nwp_lon,
+            "nwp-ecmwf__channel": nwp_channels,
+            "nwp-ecmwf__target_time_utc": nwp_time_coords,
+            "nwp-ecmwf__init_time_utc": (["nwp-ecmwf__target_time_utc"], nwp_init_time),
+            "nwp-ecmwf__step": (["nwp-ecmwf__target_time_utc"], nwp_steps),
+
+            # Site coordinates
+            "site__site_id": np.int32(1),
+            "site__latitude": site_lat,
+            "site__longitude": site_lon,
+            "site__capacity_kwp": 10000.0,
+            "site__time_utc": site_time_coords,
+            "site__solar_azimuth": (["site__time_utc"], site_solar_azimuth),
+            "site__solar_elevation": (["site__time_utc"], site_solar_elevation),
+            **{f"site__{k}": (["site__time_utc"], v) for k, v in trig_features.items()}
+        }
+    )
+
+    # Add NWP attributes
+    site_data_ds["nwp-ecmwf"].attrs = {
+        "Conventions": "CF-1.7",
+        "GRIB_centre": "ecmf",
+        "GRIB_centreDescription": "European Centre for Medium-Range Weather Forecasts",
+        "GRIB_subCentre": "0",
+        "institution": "European Centre for Medium-Range Weather Forecasts"
+    }
+
+    return site_data_ds
+
+
+def generate_synthetic_pv_batch():
+    """
+    Generate a synthetic PV batch for SimpleLearnedAggregator tests
+    """
+    # 3D tensor of shape [batch_size, sequence_length, num_sites]
+    batch_size = 8
+    sequence_length = 180 // 5 + 1
+    num_sites = 349
+
+    return torch.rand(batch_size, sequence_length, num_sites)
+
+
 @pytest.fixture()
 def sample_train_val_datamodule():
-    # duplicate the sample batcnes for more training/val data
-    n_duplicates = 10
-
+    """
+    Create a DataModule with synthetic data files for training and validation
+    """
     with tempfile.TemporaryDirectory() as tmpdirname:
-        os.makedirs(f"{tmpdirname}/train")
-        os.makedirs(f"{tmpdirname}/val")
+        # Create train and val directories
+        os.makedirs(f"{tmpdirname}/train", exist_ok=True)
+        os.makedirs(f"{tmpdirname}/val", exist_ok=True)
+
+        # Generate and save synthetic samples
+        for i in range(10):
+            sample = generate_synthetic_sample()
+            torch.save(sample, f"{tmpdirname}/train/{i:08d}.pt")
+            torch.save(sample, f"{tmpdirname}/val/{i:08d}.pt")
+
+        # Define DataModule with temporary directory
+        dm = DataModule(
+            configuration=None,
+            sample_dir=tmpdirname,
+            batch_size=2,
+            num_workers=0,
+            prefetch_factor=None,
+            train_period=[None, None],
+            val_period=[None, None],
+        )
+
+        yield dm
 
-        file_n = 0
 
-        for file_n, file in enumerate(
-            glob.glob("tests/test_data/presaved_samples_uk_regional/train/*.pt")
-        ):
-            sample = torch.load(file)
+@pytest.fixture()
+def sample_datamodule():
```

Comment:
is this the same as sample_train_val_datamodule, could then just do
```
def sample_datamodule(sample_train_val_datamodule):
    yield sample_train_val_datamodule
```

Was it the same before these changes?